### PR TITLE
zh-CN: Merge some contents, translate some texts and fix usages of some punctuations

### PR DESCRIPTION
**Pull request by TheBlueAlgae**
Part of #324.

### CODE REVIEW

**Comment by TheBlueAlgae** [comment]
I do not know why but after syncing from this repository, I found these commits go ahead of this repo. So I made this pull request.

**Comment by mgeisler** [comment]
Hey @TheBlueAlgae, please name the PRs so they start with `zh-CN:` — the PR title goes into the permanent Git history and so we can easily know which commits belong to which translation.

> I do not know why but after syncing from this repository, I found these commits go ahead of this repo. So I made this pull request.

I am not sure why this happened, but I'll take a look.

**Comment by mgeisler** [comment]
It seems that some of the commits are from a review by @emmali01:

image

**Comment by mgeisler** [comment]
I'm not sure why the change showed up this way, but they look good to me: in particular, fixing the escaped links (`\\[` in the PO file) is good since they would have resulted in bad output.

**Comment by mgeisler** [comment]
You can perhaps ask in #324 or somewhere else if you coordinate the Simplified Chinese translation in some forum?

File: po/zh-CN.po [hunk 1]
```diff
@@ -6527,24 +6617,6 @@ msgid ""
 "}\n"
 "```"
 msgstr ""
-"```rust,editable\n"
```
Comment (NoahDragon) [inline po/zh-CN.po:6530]: Curious why the following removed? But I saw some code snippets have added in above changes.

**Review by NoahDragon — CHANGES_REQUESTED** [review]

File: po/zh-CN.po [hunk 2]
```diff
@@ -1112,8 +1112,8 @@ msgid ""
 "(embedded) development. Both microcontrollers and application processors are "
 "covered."
 msgstr ""
-"[Bare-metal](bare-metal.md)：为期一天的课程，介绍如何使用 Rust 进行裸机（嵌入"
-"式）开发。课程内容涵盖微控制器和应用处理器。"
+"[裸金属](bare-metal.md)：为期一天的课程，介绍如何使用 Rust 进行裸机（嵌入式）"
```
[the remark this answers is not in the document]
Comment (TheBlueAlgae) [reply po/zh-CN.po:1115]: I searched on the Internet and found that most of companies (such as Tencent, Alibaba and Huawei) adopted "裸金属" as the translation of "bare metal", so I think this translation is more suitable than "裸机".

File: po/zh-CN.po [hunk 3]
```diff
@@ -1718,8 +1718,8 @@ msgid ""
 "Rust also has [editions](https://doc.rust-lang.org/edition-guide/): the "
 "current edition is Rust 2021. Previous editions were Rust 2015 and Rust 2018."
 msgstr ""
-"Rust 也有三个\\[版本\\]：当前版本是 Rust 2021。之前的版本是 Rust 2015 和 "
-"Rust 2018。"
+"Rust 也有三个[版本](https://doc.rust-lang.org/edition-guide/)：当前版本是 "
+"Rust 2021。之前的版本是 Rust 2015 和 Rust 2018。"
```
[the remark this answers is not in the document]
Comment (TheBlueAlgae) [reply po/zh-CN.po:1722]: Maybe using a comma is better?

File: po/zh-CN.po (hunk 1 above)
Comment (TheBlueAlgae) [reply po/zh-CN.po:6530]: It might be a mistake. I will check them later.

**Review by NoahDragon — CHANGES_REQUESTED** [review]
Thank for the PR. 

I saw there are more changes added in between reviews, does it mean this PR is still WIP and not ready for review? If so, could you please mark it in the title, so we could review it when the changes are stablized?

**Review by SketchK — APPROVED** [review]

File: po/zh-CN.po (hunk 2 above)
[the remark this answers is not in the document]
Comment (TheBlueAlgae) [reply po/zh-CN.po]: Ok, I later searched more details of "bare metal" in different fields and decided to change all of translations from "裸金属" to "裸机".

**Pull request closed without merging**